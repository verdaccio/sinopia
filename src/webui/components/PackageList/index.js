import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import Divider from '@material-ui/core/Divider';

import Package from '../Package';
import Help from '../Help';
import { formatLicense } from '../../utils/package';

import classes from './packageList.scss';

export default class PackageList extends React.Component {
  static propTypes = {
    packages: PropTypes.array,
  };

  render() {
    return (
      <div className={"package-list-items"}>
        <div className={classes.pkgContainer}>
          {this.hasPackages() ? this.renderPackages(): <Help /> }
        </div>
      </div>
    );
  }

  hasPackages() {
    const {packages} = this.props;
    return packages.length > 0;
  }

  renderPackages = () => {
    return (
      <Fragment>
        {this.renderList()}
      </Fragment>
    );
  }

  renderList = () => {
    const { packages } = this.props;
    return (
      packages.map((pkg, i) => {
        const { name, version, description, time, keywords, dist, homepage, bugs } = pkg;
        const author = pkg.author;
        // TODO: move format license to API side.
        const license = formatLicense(pkg.license);
        return (
          <React.Fragment key={i}>
            {i !== 0 && <Divider></Divider>}
            <Package {...{ name, dist, version, author, description, license, time, keywords, homepage, bugs }} />
          </React.Fragment>
        );
      })
    );
  }
}
