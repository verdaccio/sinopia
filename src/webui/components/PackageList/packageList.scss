@import '../../styles/variables';
@import '../../styles/mixins';

.pkgContainer {
  margin: 0;
  padding: 0;

  li {
    @include border-bottom-default($paleNavy);
    list-style: none;

    &:last-child {
      border-bottom: none;
    }
  }

  .listTitle {
    font-weight: $font-weight-regular;
    font-size: $font-size-xl;
    margin-top: 30px;
    margin-bottom: 0;
  }
}
