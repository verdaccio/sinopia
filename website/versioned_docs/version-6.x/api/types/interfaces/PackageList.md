---
id: 'PackageList'
title: 'Interface: PackageList'
sidebar_label: 'PackageList'
sidebar_position: 0
custom_edit_url: null
---

## Indexable

â–ª [key: `string`]: [`PackageAccess`](PackageAccess.md)
