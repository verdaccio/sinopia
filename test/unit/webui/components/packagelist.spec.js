/**
 * PackageList component
 */

import React from 'react';
import { mount } from 'enzyme';
import PackageList from '../../../../src/webui/components/PackageList/index';
import { BrowserRouter } from 'react-router-dom';

describe('<PackageList /> component', () => {
  it('should load the component with no packages', () => {
    const props = {
      packages: [],
      help: true
    };
    const wrapper = mount(
      <PackageList packages={props.packages} help={props.help} />
    );
    expect(wrapper.find('Help')).toHaveLength(1);

    expect(wrapper.find('h1').text()).toEqual('No Package Published Yet');

  });

  it('should load the component with packages', () => {
    const props = {
      packages: [
        {
          name: 'verdaccio',
          version: '1.0.0',
          time: new Date(),
          description: 'Private NPM repository',
          author: { name: 'Sam' }
        },
        {
          name: 'abc',
          version: '1.0.1',
          time: new Date(),
          description: 'abc description',
          author: { name: 'Rose' }
        },
        {
          name: 'xyz',
          version: '1.1.0',
          description: 'xyz description',
          author: { name: 'Martin' }
        }
      ],
      help: false
    };

    const wrapper = mount(
      <BrowserRouter>
        <PackageList packages={props.packages} help={props.help} />
      </BrowserRouter>
    );


    expect(wrapper.find('.listTitle').text()).toContain('Available Packages');

    // package count
    expect(wrapper.find('Package')).toHaveLength(3);
    // match snapshot
    expect(wrapper.html()).toMatchSnapshot();
  });
});
