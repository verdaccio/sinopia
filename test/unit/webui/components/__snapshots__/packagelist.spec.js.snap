// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PackageList /> component should load the component with packages 1`] = `"<div class=\\"package-list-items\\"><div class=\\"pkgContainer\\"><ul class=\\"MuiList-root-128 MuiList-padding-129 package css-jpnf4t e11fsc2k10\\"><li class=\\"MuiListItem-root-132 MuiListItem-default-135 MuiListItem-gutters-140 MuiListItem-alignItemsFlexStart-137\\"><div class=\\"MuiListItemText-root-144 package-link css-9dxxs7 e11fsc2k14\\" component=\\"div\\"><span class=\\"MuiTypography-root-30 MuiTypography-subheading-37 MuiListItemText-primary-147\\"><div class=\\"MuiGrid-container-150 MuiGrid-item-151 MuiGrid-grid-xs-12-190\\"><div class=\\"MuiGrid-item-151 MuiGrid-grid-xs-true-178\\"><a class=\\"css-1jgqs7g e11fsc2k7\\" href=\\"/-/web/detail/verdaccio\\"><span class=\\"css-10n8p5f e11fsc2k8\\">verdaccio</span></a></div><div class=\\"MuiGrid-item-151 MuiGrid-grid-xs-true-178 css-s168sa e11fsc2k9\\"></div></div></span><p class=\\"MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-colorTextSecondary-63 MuiListItemText-secondary-148\\"><span class=\\"MuiTypography-root-30 MuiTypography-body1-39 css-zd063h e11fsc2k15\\">Private NPM repository</span></p></div></li><li class=\\"MuiListItem-root-132 MuiListItem-default-135 MuiListItem-gutters-140 MuiListItem-alignItemsFlexStart-137 css-11vba5x e11fsc2k13\\"><div class=\\"css-15496ft e11fsc2k5\\"><div class=\\"MuiAvatar-root-247 css-16dqgrp e11fsc2k6\\"><img alt=\\"Sam\\" src=\\"test avatar\\" class=\\"MuiAvatar-img-249\\"></div><span class=\\"css-eazh50 e11fsc2k4\\"><div class=\\"e11fsc2k3 css-16bvijn e1pneb170\\">Sam</div></span></div><span class=\\"css-1rxy2wf e11fsc2k0\\"><svg class=\\"e11fsc2k1 css-pqdkph ej4jd2o0\\"><title>Version</title><use xlink:href=\\"[object Object]#version\\"></use></svg>v1.0.0</span><span class=\\"css-1rxy2wf e11fsc2k0\\"><svg class=\\"e11fsc2k1 css-pqdkph ej4jd2o0\\"><title>Time</title><use xlink:href=\\"[object Object]#time\\"></use></svg><span class=\\"css-1dp07e1 e11fsc2k2\\">Published on 21.07.2018, 22:11:12 •</span>8 months ago</span></li></ul><hr class=\\"MuiDivider-root-250\\"><ul class=\\"MuiList-root-128 MuiList-padding-129 package css-jpnf4t e11fsc2k10\\"><li class=\\"MuiListItem-root-132 MuiListItem-default-135 MuiListItem-gutters-140 MuiListItem-alignItemsFlexStart-137\\"><div class=\\"MuiListItemText-root-144 package-link css-9dxxs7 e11fsc2k14\\" component=\\"div\\"><span class=\\"MuiTypography-root-30 MuiTypography-subheading-37 MuiListItemText-primary-147\\"><div class=\\"MuiGrid-container-150 MuiGrid-item-151 MuiGrid-grid-xs-12-190\\"><div class=\\"MuiGrid-item-151 MuiGrid-grid-xs-true-178\\"><a class=\\"css-1jgqs7g e11fsc2k7\\" href=\\"/-/web/detail/abc\\"><span class=\\"css-10n8p5f e11fsc2k8\\">abc</span></a></div><div class=\\"MuiGrid-item-151 MuiGrid-grid-xs-true-178 css-s168sa e11fsc2k9\\"></div></div></span><p class=\\"MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-colorTextSecondary-63 MuiListItemText-secondary-148\\"><span class=\\"MuiTypography-root-30 MuiTypography-body1-39 css-zd063h e11fsc2k15\\">abc description</span></p></div></li><li class=\\"MuiListItem-root-132 MuiListItem-default-135 MuiListItem-gutters-140 MuiListItem-alignItemsFlexStart-137 css-11vba5x e11fsc2k13\\"><div class=\\"css-15496ft e11fsc2k5\\"><div class=\\"MuiAvatar-root-247 css-16dqgrp e11fsc2k6\\"><img alt=\\"Rose\\" src=\\"test avatar\\" class=\\"MuiAvatar-img-249\\"></div><span class=\\"css-eazh50 e11fsc2k4\\"><div class=\\"e11fsc2k3 css-16bvijn e1pneb170\\">Rose</div></span></div><span class=\\"css-1rxy2wf e11fsc2k0\\"><svg class=\\"e11fsc2k1 css-pqdkph ej4jd2o0\\"><title>Version</title><use xlink:href=\\"[object Object]#version\\"></use></svg>v1.0.1</span><span class=\\"css-1rxy2wf e11fsc2k0\\"><svg class=\\"e11fsc2k1 css-pqdkph ej4jd2o0\\"><title>Time</title><use xlink:href=\\"[object Object]#time\\"></use></svg><span class=\\"css-1dp07e1 e11fsc2k2\\">Published on 21.07.2018, 22:11:12 •</span>8 months ago</span></li></ul><hr class=\\"MuiDivider-root-250\\"><ul class=\\"MuiList-root-128 MuiList-padding-129 package css-jpnf4t e11fsc2k10\\"><li class=\\"MuiListItem-root-132 MuiListItem-default-135 MuiListItem-gutters-140 MuiListItem-alignItemsFlexStart-137\\"><div class=\\"MuiListItemText-root-144 package-link css-9dxxs7 e11fsc2k14\\" component=\\"div\\"><span class=\\"MuiTypography-root-30 MuiTypography-subheading-37 MuiListItemText-primary-147\\"><div class=\\"MuiGrid-container-150 MuiGrid-item-151 MuiGrid-grid-xs-12-190\\"><div class=\\"MuiGrid-item-151 MuiGrid-grid-xs-true-178\\"><a class=\\"css-1jgqs7g e11fsc2k7\\" href=\\"/-/web/detail/xyz\\"><span class=\\"css-10n8p5f e11fsc2k8\\">xyz</span></a></div><div class=\\"MuiGrid-item-151 MuiGrid-grid-xs-true-178 css-s168sa e11fsc2k9\\"></div></div></span><p class=\\"MuiTypography-root-30 MuiTypography-body1-39 MuiTypography-colorTextSecondary-63 MuiListItemText-secondary-148\\"><span class=\\"MuiTypography-root-30 MuiTypography-body1-39 css-zd063h e11fsc2k15\\">xyz description</span></p></div></li><li class=\\"MuiListItem-root-132 MuiListItem-default-135 MuiListItem-gutters-140 MuiListItem-alignItemsFlexStart-137 css-11vba5x e11fsc2k13\\"><div class=\\"css-15496ft e11fsc2k5\\"><div class=\\"MuiAvatar-root-247 css-16dqgrp e11fsc2k6\\"><img alt=\\"Martin\\" src=\\"test avatar\\" class=\\"MuiAvatar-img-249\\"></div><span class=\\"css-eazh50 e11fsc2k4\\"><div class=\\"e11fsc2k3 css-16bvijn e1pneb170\\">Martin</div></span></div><span class=\\"css-1rxy2wf e11fsc2k0\\"><svg class=\\"e11fsc2k1 css-pqdkph ej4jd2o0\\"><title>Version</title><use xlink:href=\\"[object Object]#version\\"></use></svg>v1.1.0</span></li></ul></div></div>"`;
